src/chain/electrum.rs

```diff
@@ -0,0 +1,48 @@
+// This file is Copyright its original authors, visible in version control history.
+//
+// This file is licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. You may not use this file except in
+// accordance with one or both of these licenses.
+
+use crate::error::Error;
+use crate::logger::{log_error, LdkLogger, Logger};
+
+use lightning_transaction_sync::ElectrumSyncClient;
+
+use bdk_electrum::BdkElectrumClient;
+
+use electrum_client::Client as ElectrumClient;
+
+use std::sync::Arc;
+
+pub(crate) struct ElectrumRuntimeClient {
+	electrum_client: Arc<ElectrumClient>,
+	bdk_electrum_client: Arc<BdkElectrumClient<ElectrumClient>>,
+	tx_sync: Arc<ElectrumSyncClient<Arc<Logger>>>,
+	runtime: Arc<tokio::runtime::Runtime>,
+	logger: Arc<Logger>,
+}
+
+impl ElectrumRuntimeClient {
+	pub(crate) fn new(
+		server_url: String, runtime: Arc<tokio::runtime::Runtime>, logger: Arc<Logger>,
+	) -> Result<Self, Error> {
+		let electrum_client = Arc::new(ElectrumClient::new(&server_url).map_err(|e| {
```

Comment:
You mention in the PR desc that conn reestablish is missing still. How would ldk-node behave when that happens?

Reply:
Could this be a reason to not yet merge? Perhaps a hard exit for now could help avoid users not being aware that they aren't watching the chain anymore.

Reply:
Can understand you'd rather not wait, but does seem important to know and document what's going to happen in the mean time before merging this PR?